include/socket_handler.h

```diff
@@ -148,41 +227,14 @@ class SocketHandler {
     size_t amount_read = readBuffer(buffer, kBufferSize);
     std::string http_string(buffer, amount_read);
 
-    size_t http_index = 0;
-    std::string method = substringToken(http_index, http_string, " ");
-    if (method != "GET" && method != "POST") {
+    ParseMethodAndEndpointFromHttpRequest(http_string, &request);
+    ParseHeadersFromHttpRequest(http_string, &request);
+
+    if (request.method == http::Method::kUnsupported) {
       throw std::invalid_argument(
           "Invalid HTTP Request: Only support GET and POST!");
     }
-
-    std::string endpoint_and_args =
-        substringToken(http_index, http_string, " ");
-    std::unordered_map<std::string, std::string> args;
-    parseArgs(endpoint_and_args, args);
-    request.endpoint = endpoint_and_args;
-    request.args = args;
-
-    if (method == "GET") {
-      request.method = http::kGet;
-      this->respondOK();
-      return request;
-    }
-    request.method = http::kPost;
-
-    substringToken(
-        http_index, http_string,
-        "Content-Type: ");  // Technically headers are case-insenstive but
-                            // practice is that they are supposed to be
-                            // capatilized can be fixed later!
-    request.args["content_type"] =
-        substringToken(http_index, http_string, "\n");
-
-    http_index = 0;
-    substringToken(http_index, http_string, "Content-Length: ");
-    std::string str_size = substringToken(http_index, http_string, "\n");
-    unsigned long long size = strtoul(str_size.c_str(), NULL, 10);
-    request.args["file_name"] = readBinaryData(size, http_string);
```

Comment:
this will be added back in the future, we will have dedicated lambdas to handle different endpoints and thats where `readBinaryData` eventually will live